### DISCUSSION TITLE
CC mapping is controlling unwanted targets

### DISCUSSION THREAD

**Discussion by ilikemyrealname** [helgoboss/helgobox#453] [Q&A]
It seems after creating about 6-10 mappings that some start to control targets outside of what's intended.

I'm trying to use the encoders on my keyboard to control plugin parameters. The goal is to map all my most used plugins. For proof of concept, I started with Waves PuigTec EQP1A and MEQ5. I'm able to map most of their parameters and they work flawlessly, but after so many it seems the mapping starts going haywire. I'll go back to previously working mappings and it'll control the parameter I intended, but then also control an addition parameter at the same time. Once that happens, even if I remove the mapping, the unintended target will still move when I turn the encoder. Then I have to start over and try again. 

I'm new to Reaper and ReaLearn, so forgive me if this is user error.

**Comment by helgoboss** [comment]
Please post the RPP. Best without 3rd-party plugins.

**Reply by ilikemyrealname** [reply]
Thank you for the reply!

So I should expect there to be issues trying to control Waves or any of my other plugins? Is there another solution for controlling plugins within Reaper then? I tried the built in Learn function, but lack of feedback meant parameters would get messed up when working between multiple plugins.

Link to file:
https://drive.google.com/file/d/12uh31WRZ1izM2I-fCR_HCwtXOUIYnJX7/view?usp=sharing

**Reply by ilikemyrealname** [reply]
> Please post the RPP. Best without 3rd-party plugins.

Same thing is happening with 1st-party plugins. Again, I apologize if this is user error. I attached a link to the rpp file that has the mappings in question.

https://drive.google.com/file/d/1b4kiWCvalL2DBxU1PHeIdT2F8pWKqYNK/view?usp=sharing

**Reply by helgoboss** [reply]
Yes, those mappings probably don't do what you think they do. Let's look at the group that you have named "Cockos ReaXcomp". The first mapping in particular, which is named "24". The target section of that mapping tells ReaLearn: "Set the value of the first parameter in the currently focused FX!". So no matter which plug-in you focus, ReaLearn will let you control its first parameter. That's it. It doesn't magically check if the plug-in type is ReaXcomp (the group name is totally irrelevant here, just in case you thought that this is making the difference). So it will apply to any focused plug-in.

Is it possible you want to do this? [Tutorial 6.3: Using "Auto-load" to control whatever plug-in is currently in focus](https://github.com/helgoboss/realearn/blob/master/doc/user-guide.adoc#using-auto-load-to-control-whatever-plug-in-is-currently-in-focus) (you might need to scroll a bit down)

**Reply by ilikemyrealname** [reply]
You are correct, that is where I was mistaken. Following the tutorial you shared has me up and running.

This is an incredible piece of software, really amazing. I'm coming from 15 years of Pro Tools so the level of freedom is jaw dropping.

**Comment by helgoboss** [comment]
But be aware, "Auto-load preset" has some problems with shell plug-ins (see this issue: #432). I think Waves plug-ins are always shell plug-ins. The result: ReaLearn is currently not able to distinguish different Waves plug-in types and will therefore always load the same main preset. I need to sort out that one.

**Reply by ilikemyrealname** [reply]
Now that I understand how Auto-load works, this makes sense. If there is anything I can do non-programming-wise to aid with the Waves stuff, I'm willing to help.
